pandas/tests/indexes/test_multi.py

```diff
@@ -154,7 +154,7 @@ def assert_matching(actual, expected):
             # as much as possible
             self.assertEqual(len(actual), len(expected))
             for act, exp in zip(actual, expected):
-                act = np.asarray(act)
+                act = np.asarray(act, dtype=np.object_)
```

Comment:
It looks like maybe `_ensure_index` is doing the conversion to `np.object_` in that test case (https://github.com/pydata/pandas/blob/master/pandas/indexes/base.py#L3616 and https://github.com/pydata/pandas/blob/master/pandas/lib.pyx#L1029).